Makefile

```diff
@@ -20,17 +22,26 @@ test-package-files:
 		! ./bin/md5-hash $$pkg | grep -qv $(SOURCE_PKG_MD5) || { echo "$$pkg does not match main package.json"; exit 1; }; \
 	done
 
-test-assignment:
+copy-assignment:
 	@cp package.json exercises/$(ASSIGNMENT)
-	@echo "running tests for: $(ASSIGNMENT)"
 	@mkdir -p $(OUTDIR)
 	@cp exercises/grains/lib/big-integer.$(FILEEXT) $(OUTDIR)
 	@cp exercises/$(ASSIGNMENT)/$(TSTFILE) $(OUTDIR)
 	@cp exercises/$(ASSIGNMENT)/$(EXAMPLE) $(OUTDIR)/$(subst _,-,$(ASSIGNMENT)).$(FILEEXT)
-	@sed 's/xtest/test/g' exercises/$(ASSIGNMENT)/$(TSTFILE) > $(OUTDIR)/temp.$(TSTFILE)
-	@node_modules/.bin/jest $(OUTDIR)/temp.*.spec.js
-	@rm $(OUTDIR)/*
-	@rmdir $(OUTDIR)
+	@sed 's/xtest/test/g' exercises/$(ASSIGNMENT)/$(TSTFILE) > tmp_exercises/$(ASSIGNMENT)/$(TSTFILE)
+
+# To be run as: make test-assignment ASSIGNMENT=hello-world
+test-assignment:
+	$(MAKE) -s copy-assignment
+	@node_modules/.bin/jest $(OUTDIR)
+	@rm -rf $(OUTDIR)
+
+test-travis:
+	@echo "Preparing tests..."
+	@for assignment in $(ASSIGNMENTS); do ASSIGNMENT=$$assignment $(MAKE) -s copy-assignment || exit 1; done
```

Comment:
Could lines 41 to 43 be replaced by a single line that uses the `test-assignment` task?

```
@for assignment in $(ASSIGNMENTS); do ASSIGNMENT=$$assignment $(MAKE) -s test-assignment || exit 1; done
```

Otherwise, I don't see any usage of `test-assignment`.